fighters/snake/src/status.rs

```diff
@@ -0,0 +1,737 @@
+        let entry_id = WorkModule::get_int(fighter.module_accessor, *FIGHTER_INSTANCE_WORK_ID_INT_ENTRY_ID) as usize;
+        if ControlModule::check_button_on(fighter.module_accessor, *CONTROL_PAD_BUTTON_CATCH)
+        || ControlModule::check_button_on(fighter.module_accessor, *CONTROL_PAD_BUTTON_GUARD) {
+            if StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_AIR {
+                fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_DROP.into(), false.into());
+            }
+            else {
+                fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_ESTABLISH_GROUND.into(), false.into());
+            }
+            VarModule::on_flag(fighter.object(), vars::snake::instance::SELF_STICK);
+        }
+        else {
+            VarModule::off_flag(fighter.object(), vars::snake::instance::SELF_STICK);
+            fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_ESTABLISH_JUDGE.into(), false.into());
+        }
+        false.into()
+    }
+    else {
+        fun_7100018800(fighter, false);
+        true.into()
+    }
+}
+pub unsafe fn fun_7100018800(fighter: &mut L2CFighterCommon, skip_check: bool) {
+    if skip_check == false {
+        if StatusModule::situation_kind(fighter.module_accessor) == StatusModule::prev_situation_kind(fighter.module_accessor) {
+            return
+        }
+    }
+    if StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_GROUND {
+        GroundModule::set_rhombus_offset(fighter.module_accessor, &Vector2f{x:0.0, y:0.0});
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_GROUND_STOP);
+        fighter.set_situation(SITUATION_KIND_GROUND.into());
+        GroundModule::correct(fighter.module_accessor, GroundCorrectKind(*GROUND_CORRECT_KIND_GROUND));
+        let motion = WorkModule::get_int64(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_WORK_INT_MOT_KIND);
+        MotionModule::change_motion_inherit_frame(fighter.module_accessor, Hash40::new_raw(motion), -1.0, 1.0, 0.0, false, false);
+    }
+    else {
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_FALL);
+        fighter.set_situation(SITUATION_KIND_AIR.into());
+        GroundModule::correct(fighter.module_accessor, GroundCorrectKind(*GROUND_CORRECT_KIND_AIR));
+        let motion = WorkModule::get_int64(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_WORK_INT_MOT_AIR_KIND);
+        MotionModule::change_motion_inherit_frame(fighter.module_accessor, Hash40::new_raw(motion), -1.0, 1.0, 0.0, false, false);
+    }
+}
+
+#[status_script(agent = "snake", status = FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_PRODUCE, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_END)]
+unsafe fn snake_down_special_start_status_end(fighter: &mut L2CFighterCommon) -> L2CValue {
+    if fighter.global_table[0xb].get_i32() != *FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_ESTABLISH_GROUND
```

Comment:
We have actual names for the constants, this one is probably `PREV_STATUS_KIND`, you can check `dynamic/src/constants.rs`